### PR TITLE
Update dask interface during XArray integration

### PR DESCRIPTION
**Pull request by mrocklin**
This PR contains some fixes to Dask's new interface API as I add functionality to XArray

This builds on @jcrist 's work in https://github.com/dask/dask/pull/2748

### CODE REVIEW

**Comment by mrocklin** [comment]
Co-developing with https://github.com/pydata/xarray/pull/1674

File: dask/base.py [hunk 1]
```diff
@@ -36,6 +36,8 @@ def is_dask_collection(x):
         return x.__dask_graph__() is not None
     except AttributeError:
         return False
+    except TypeError:
+        return False
```
Comment (jcrist) [inline dask/base.py]:
```python
except (AttributeError, TypeError):
    return False
```

File: dask/tests/test_base.py [hunk 2]
```diff
@@ -627,3 +627,8 @@ def my_get(dsk, keys):
 
     with dask.set_options(array_optimize=None, get=my_get):
         y.compute()
+
+
+def test_class_is_not_collection():
+    da = pytest.importorskip('dask.array')
+    assert not is_dask_collection(da.Array)
```
Comment (jcrist) [inline dask/tests/test_base.py]: Would prefer to just add this to the bottom of `test_is_dask_collection` above:

```
assert not is_dask_collection(DummyCollection)
```

Comment (mrocklin) [reply dask/tests/test_base.py]: Fixed

File: dask/base.py (hunk 1 above)
Comment (mrocklin) [reply dask/base.py]: Fixed

**Comment by mrocklin** [comment]
@jcrist ok to merge?

**Comment by jcrist** [comment]
Looks good to me. Merging.

**Pull request merged**